VersionControlUI/src/commands/CDiff.cpp

```diff
@@ -45,50 +40,152 @@ using namespace FilePersistence;
 
 namespace VersionControlUI {
 
-static const QString overlayGroupName{"DiffHighlights"};
+CDiff::CDiff() : Command{"diff"} {}
 
-CDiff::CDiff() : CommandWithFlags{"diff", {{"project"}}, true, false}
-{}
+bool CDiff::canInterpret(Visualization::Item*, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& )
+{
+	QString managerName = target->node()->manager()->name();
+
+	QStringList commandTokensCopy = commandTokens;
+
+	// get GitRepository
+	QString path{"projects/" + managerName};
+
+	if (GitRepository::repositoryExists(path))
+	{
+		GitRepository repository{path};
 
-Interaction::CommandResult* CDiff::executeNamed(Visualization::Item*, Visualization::Item* target,
-											  const std::unique_ptr<Visualization::Cursor>&,
-											  const QString& name, const QStringList&)
+		// if there are no tokens we are unable to interpret the command
+		if (commandTokensCopy.isEmpty())
+			return false;
+
+		// check that command name starts with the characters of the first token
+		if (!name().startsWith(commandTokensCopy.takeFirst()))
+			return false;
+
+		// if there is an additional version specified, check that it is a valid commit
+		if (!commandTokensCopy.isEmpty() && !repository.isValidRevisionString(commandTokensCopy.takeFirst()))
+			return false;
+
+		// check the same if there is a second version specified
+		if (!commandTokensCopy.isEmpty() && !repository.isValidRevisionString(commandTokensCopy.takeFirst()))
+			return false;
+
+		return true;
+	}
+	else return false;
+}
+
+Interaction::CommandResult* CDiff::execute(Visualization::Item*, Visualization::Item* target,
+				const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& )
 {
 	auto scene = target->scene();
 	scene->clearFocus();
 	scene->clearSelection();
 	scene->setMainCursor(nullptr);
 
-	Model::TreeManager* headManager = target->node()->manager();
-	QString managerName = headManager->name();
+	QString managerName = target->node()->manager()->name();
+
 
-	// TODO add functionality to specify two versions
-	VersionControlUI::DiffManager diffManager{name, "HEAD", managerName, Model::SymbolMatcher{"Class"}};
+	// TODO restrict versions to versionA always be older?
+	// TODO diff with no versions should compare HEAD to workingDir if possible
```

Comment:
I believe, you should just `WORKDIR` instead of `HEAD` in one of the versions below. Check the `Diff.cpp` code to be sure.